plugin.video.dailymotion_com/default.py

```diff
@@ -0,0 +1,629 @@
+        return
+    else:
+        cc= content['qualities']  #['380'][0]['url']
+
+        cc = cc.items()
+        cc = sorted(cc, reverse=True)
+ 
+        m_url = ''
+        other_playable_url = []
+
+        for source,json_source in cc:
+
+            for item in json_source:
+                m_url = item.get('url',None)
+                xbmc.log("DILYMOTION_Gates - m_url = %s" %m_url)
+                if m_url:
+                    if not live:
+
+                        if source == "auto" :
+                            continue
+
+                        elif  int(source) <= int(maxVideoQuality) :
+                            rr = requests.get(m_url,cookies=r.cookies.get_dict() ,headers=headers)    
+                            inhalt=rr.text.split('#cell')[0]                          
+                            match = re.compile('http://(.+?).m3u8', re.DOTALL).findall(inhalt)
+                            return "http://"+match[0]+".m3u8"
+
+                        elif '.mnft' in m_url:
+                            continue
+                         
+                    else:
+                        if '.m3u8?auth' in m_url:
+                            m_url = m_url.split('?auth=')
+                            the_url = m_url[0] + '?redirect=0&auth=' + urllib.quote(m_url[1])
+                            rr = requests.get(the_url,cookies=r.cookies.get_dict() ,headers=headers)
+                            if rr.headers.get('set-cookie'):
+                                print 'adding cookie to url'
+                                return rr.text.split('#cell')[0]+'|Cookie='+rr.headers['set-cookie']
+                            else:
+                                return rr.text.split('#cell')[0]
+                    other_playable_url.append(m_url) 
+        if len(other_playable_url) >0: # probable not needed only for last resort
+            for m_url in other_playable_url:
+                if '.m3u8?auth' in m_url:
+                    sep_url = m_url.split('?auth=')
+                    the_url = sep_url[0] + '?redirect=0&auth=' + urllib.quote(sep_url[1])
+                    rr = requests.get(the_url,cookies=r.cookies.get_dict() ,headers=headers)
+                    if rr.headers.get('set-cookie'):
+                        print 'adding cookie to url'
+                        return rr.text.split('#cell')[0]+'|Cookie='+rr.headers['set-cookie']
+                    else:
+                        return rr.text.split('#cell')[0]
+                    
+      
+
+
+def queueVideo(url, name):
+    playlist = xbmc.PlayList(xbmc.PLAYLIST_VIDEO)
+    listitem = xbmcgui.ListItem(name)
+    playlist.add(url, listitem)
+
+
+def downloadVideo(id):
+    downloader = SimpleDownloader.SimpleDownloader()
+    content = getUrl2("http://www.dailymotion.com/embed/video/"+id)
+    match = re.compile('<title>(.+?)</title>', re.DOTALL).findall(content)
+    global downloadDir
+    while not downloadDir:
```

Comment:
also here: please dont put addon.OpenSettings() into a loop. that could be really confusing for users.